sys/linux/binfmt.txt

```diff
@@ -3,15 +3,27 @@
 
 include <uapi/linux/a.out.h>
 include <uapi/linux/elf.h>
+include <linux/fcntl.h>
 
-execve(file ptr[in, filename], argv ptr[in, array[ptr[in, string]]], envp ptr[in, array[ptr[in, string]]])
-execveat(dirfd fd_dir, file ptr[in, filename], argv ptr[in, array[ptr[in, string]]], envp ptr[in, array[ptr[in, string]]], flags flags[at_flags])
+resource fd_binfmt[fd]
+resource ptr_binfmt_file[intptr]
 
-write$binfmt_script(fd fd, data ptr[in, binfmt_script], len bytesize[data])
-write$binfmt_misc(fd fd, data ptr[in, binfmt_misc], len bytesize[data])
-write$binfmt_aout(fd fd, data ptr[in, binfmt_aout], len bytesize[data])
-write$binfmt_elf32(fd fd, data ptr[in, binfmt_elf32], len bytesize[data])
-write$binfmt_elf64(fd fd, data ptr[in, binfmt_elf64], len bytesize[data])
+syz_create_resource$binfmt(file ptr[in, filename]) ptr_binfmt_file
```

Comment:
I remember there was some check that different variants of the same syscall must have similar argument types.

In this particular case, will it also tolerate the descriptions of the form `syz_create_resource$abcd(x int32)` / `syz_create_resource$abcd(x int64)`?